### PR TITLE
Force kube 1.14 golang.org/x/net

### PR DESCRIPTION
**Pull request by enj**
This allows the http2 code to respect "Connection" == "close" to mean sending a GOAWAY and tearing down the TCP connection.

Signed-off-by: Monis Khan <mkhan@redhat.com>

@openshift/sig-auth @openshift/sig-master 

@smarterclayton I believe you said you wanted this to be bumped to latest?

### CODE REVIEW

**Comment by smarterclayton** [comment]
Bumped to kube, preferably.

**Comment by enj** [comment]
> Bumped to kube, preferably.

Set it to `release-branch.go1.12`

File: glide.yaml [hunk 1]
```diff
@@ -176,6 +176,9 @@ import:
 # auth (for sally)
 - package: github.com/gorilla/sessions
   version: a3acf13e802c358d65f249324d14ed24aac11370
+# auth (for sally)
```
Comment (deads2k) [inline glide.yaml]: good thing we all love sally

**Comment by deads2k** [comment]
So uh, for those of us in the peanut gallery, is there a thing you need to do to trigger this?  You definitely need to at least try to make test proving you destroy these connections and if you have code to add to make this bump effective, you need that too.  I'm not seeing a good reason to do this bump yet.

**Comment by tnozicka** [comment]
There were some issues upstream in `net` package, not sure if we hook up the one from golang or the experimental to those particular places. (e.g. https://github.com/golang/go/issues/27044)

This would also mean we take pretty significant change from upstream that is compiled and tested with different go version. And there are regressions if we consider bumping to go1.12 - https://github.com/kubernetes/kubernetes/issues/75833 https://github.com/kubernetes/kubernetes/issues/75837

**Comment by enj** [comment]
#22529

**Pull request closed without merging**